src/app/server-cluster/StandardServerCluster.h

```diff
@@ -0,0 +1,88 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/server-cluster/ServerClusterInterface.h>
+
+namespace chip {
+namespace app {
+
+/// Provides an implementation of most methods for a `ServerClusterInterface`
+/// to make it easier to implement spec-compliant classes.
+///
+/// In particular it does:
+///   - maintains a data version and provides `IncreaseDataVersion`. Ensures this
+///     version is spec-compliant initialized (with a random value)
+///   - Provides default implementations for most virtual methods EXCEPT:
+///       - ReadAttribute (since that one needs to handle featuremap and revision)
+///       - GetClusterId (since every implementation is for different clusters)
+///
+///
+class StandardServerCluster : public ServerClusterInterface
+{
+public:
+    StandardServerCluster();
+    virtual ~StandardServerCluster() = default;
+
+    StandardServerCluster(const StandardServerCluster & other)             = default;
+    StandardServerCluster(StandardServerCluster && other)                  = default;
+    StandardServerCluster & operator=(const StandardServerCluster & other) = default;
+    StandardServerCluster & operator=(StandardServerCluster && other)      = default;
+
+    void IncreaseDataVersion() { mDataVersion++; }
+
+    //////////////////////////// ServerClusterInterface implementation ////////////////////////////////////////
+
+    DataVersion GetDataVersion() const override { return mDataVersion; }
+    BitFlags<DataModel::ClusterQualityFlags> GetClusterFlags() const override;
+
+    /// Default implementation errors out with an unsupported write on every attribute.
+    DataModel::ActionReturnStatus WriteAttribute(const DataModel::WriteAttributeRequest & request,
+                                                 AttributeValueDecoder & decoder) override;
+
+    /// Must only be implemented if support for any non-global attributes
+    /// is required.
+    ///
+    /// Default implementation just returns the above global attributes.
```

Comment:
This could use some discussion about how a subclass could/should implement this.  For example, can it add all its non-global attributes to the builder and then call this superclass method to add the global ones?  If it can, we should say that, because that seems like a good idea for subclasses to do (though of course they don't have to).  If it can't, we should point that out and say what it should generally do instead.

Reply:
I was torn here: we could provide this as it would be 100% reusable. However then I have to always return this list in RAM where as the current implementation allows for in-flash return. The global attributes contain 5 items and if we pack the content, they take 8 bytes each, so the overhead in flash would be 40 bytes per cluster in flash if we decide not to use RAM at all.

So my choice was:
   - let subclasses figure it out (like right now) and then we hard-code everything including extra 40 bytes overhead for every implemented cluster
   - provide convenience "append" methods
   
I opted to not provide the append since I can always add it later however once added I cannot remove.